src/Formatter/Formatter.php

```diff
@@ -227,4 +250,69 @@ public function getJs()
     {
         return $this->getComponent('js');
     }
+
+    protected function ensureRendererExists()
+    {
+        if (! static::$formatter) {
+            return;
+        }
+
+        $revision = $this->getRevision();
+
+        if (file_exists($this->cacheDir."/Renderer_$revision.php")) {
+            return;
+        }
+
+        $renderer = Arr::get(static::$formatter, 'renderer');
+
+        if (! empty($renderer)) {
+            file_put_contents($this->cacheDir."/Renderer_$revision.php", $renderer);
+        } else {
+            $this->finalize();
+        }
+
+        // Reload because finalizing might have generated a new one.
+        $renderer = Arr::get(static::$formatter, 'renderer');
+
+        if ($renderer && static::$formatter['renderer'] instanceof __PHP_Incomplete_Class) {
+            // Autoload the file from disk using a simple include, while suppressing errors.
+            @include $this->cacheDir."/Renderer_$revision.php";
```

Comment:
If the renderer is an incomplete class, why would loading it in again help? I'm not sure what the goal of this whole `if` statement is.